### ISSUE TITLE
Custom Widget Communication

### ISSUE THREAD

**Issue by AlexTukalskiy-CRL** [Azure/api-management-developer-portal#2194]
I have own custom widget to show APIs list. To view page with my custom widget I need to login to the Dev portal as a Developer.
But when I click on any of links (for example <a class="btn btn-secondary" target="_top" href="https://contoso.com/api-details#api=SomeAPI">Some API</a>) from my widget and set the location the top browser page I lost my auth cookies and go to the API Details page as Guest and not as Developer.

I use Managed Developer Portal.

**Comment by devagarwal007** [comment]
Hey @AlexTukalskiy-CRL, could you please help me to understand how to create these kind of custom widgets. I want to create my own custom widget for product, APIs and operations

**Comment by AlexTukalskiy-CRL** [comment]
@devagarwal007 Almost all necessary information will be provided after you created a Custom Widget.

Follow the steps below to create, implement, and deploy a custom widget.
[Learn more](https://aka.ms/apimdocs/portal/customwidgets)

**Comment by devagarwal007** [comment]
Hey @AlexTukalskiy-CRL , thankyou for your response,
I am not getting the values of product, APIs and operations directly in to the custom widget code. Also not mentioned in this documentation. Could you please help ho to get the values of product APIs and operations in the custom widget code?

**Comment by devagarwal007** [comment]
Are you talking about fetching the APIs, operations, and product values from APIM rest APIs
https://learn.microsoft.com/en-us/rest/api/apimanagement/

**Comment by AlexTukalskiy-CRL** [comment]
@devagarwal007 Yes. all Products, APIs, Operations and etc. should be retrived by REST requests to the APIM

**Comment by versoftrd** [comment]
Hello. I have this exact Issue said by @AlexTukalskiy-CRL . Any update? Thanks in advance
